Jedimaster4559/CS499-DartBoard
# Initial Project Setup and Main Component Demos

# Overview
This update adds an initial implementation of our directory structure including demos of each of the basic features. This update also adds updates for our GitHub Actions workflow.

# Details
- Adds @ethanshattuck's demo for PySide2.0
- Adds basic database support using django
- PySide2.0 Demo updates the django database
- Testing
  - I added a basic unit test to give a short demo on how this works. As we write features that can be unit tested, please include those tests here.
- Workflows
  - I've added a basic github workflow that is broken into three main stages:
    - Testing: Runs the unit tests and passes or fails based off of those results. I have set this stage to be required for merging pull requests
    - Build: This builds the code using `pyinstaller`. This is currently broken in the pipeline. See #2.  This stage can be imitated locally using the `build_executable.bat` script on your local machine.
    - Documentation: This stage builds a documentation site off of the javadocs. Right now, it just uploads the zipped up site but I eventually plan to integrate this with some other tools including GitHub pages to provide a more thorough documentation website that is more readily available.
 - Documentation
   - General Project readme has been updated to include documentation on how to get started with development
   - a `docs/` folder has been added for high level documentation. Please use this for documenting anything that shouldn't go directly with the code. Eventually, this will also be built into our site using our Documentation job.
 - Misc
   - Added `CODEOWNERS` for automated PR notification.
   - Added generic python `.gitignore`

# Testing
Please complete the following steps while testing this PR:
- Pull down  the repository/branch and run through all of the installation instructions in the `README.md`
- Ensure that running `dartboard_main.py` directly works on your dev machine.
  - Note that a `db.sqlite3` file should be automatically generated
  - Note that a list of points should appear in the console as you click. This list should get longer as you go.
-  Close the application, then restart it. Make sure that the list of points was persistent by continuing to click.
- Run the `build_executable.bat` script and ensure that the packaged executable in your `dist/` directory works on your local machine
- Read through the code and ensure that it meets quality standards and everything makes sense.
